src/array.jl

```diff
@@ -463,7 +463,7 @@ function merge_pools!(A::CatArrOrSub,
                       updaterefs::Bool=true,
                       updatepool::Bool=true)
     if isordered(A) && length(pool(A)) > 0 && pool(B) ⊈ pool(A)
-        lev = A isa CategoricalValue ? get(B) : first(setdiff(levels(B), levels(A)))
+        lev = B isa CategoricalValue ? unwrap(B) : first(setdiff(levels(B), levels(A)))
```

Comment:
@nalimilan - I think there was a bug here as `B isa` test should be made. However, since it was not caught by tests can you please review it (and possibly suggest a test that would catch this problem)? Thank you!

Reply:
bump

Reply:
Funny. So actually the first part of the branch wouldn't be appropriate anyway now so it's good it was never taken. :-) We add all levels from `B` to `A`, so printing `B` isn't very relevant as the problem may be due to other levels.

The best fix would probably be to extend `OrderedLevelsException` to take all values in `setdiff(levels(B), levels(A))`. But for this PR you can just do `lev = first(setdiff(levels(B), levels(A)))`, with a TODO.

Reply:
I have added a TODO comment as you understand better what is going on (and I assume you are rewriting levels merging code in #331 so it can be fixed there - right?)